microcosm_flask/memory.py

```diff
@@ -0,0 +1,41 @@
+from logging import Logger
+from tracemalloc import start, take_snapshot
+
+from microcosm.api import binding, defaults, typed
+from microcosm.config.types import boolean
+
+
+@binding("memory_profiler")
+@defaults(
+    enabled=typed(boolean, False),
+    report_size_lines=typed(int, 10),
+)
+class MemoryProfiler:
+    logger: Logger
+
+    def __init__(self, graph):
+        self.enabled = graph.config.memory_profiler.enabled
+        self.report_size_lines = graph.config.memory_profiler.report_size_lines
+        self.logger = graph.logger
+
+        if not self.enabled:
+            self.logger.info("Skipping initialization because memory profiling is not enabled!")
+            return
+
+        start()
+        self.origin_snapshot = take_snapshot()
+
+    def take_snapshot(self, func):
```

Comment:
I ebbed and flowed on using a decorator, but that's what fits the pattern we have for adding this behavior on a route in mc-flask. We can change/generalize if there's ever a use case.